### PR TITLE
Nuke EKS clusters

### PR DESCRIPTION
**Pull request by yorinasub17**
Given how expensive EKS is (one cluster = $0.20/hr), and how frequent the tests fail right now as I actively develop, I am afraid I might forget to clean up one of these so decided to go ahead and implement the nuking.

Note that this required upgrading `aws-sdk-go`. I tried the latest and there were issues, so downgraded to what we have in terratest.

### CODE REVIEW

File: aws/eks_utils_for_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+func getRandomEksSupportedRegion() string {
+	supportedRegions := []string{
+		"us-east-1",
+		"us-east-2",
+		"us-west-2",
+		"eu-west-1",
+	}
+	rand.Seed(time.Now().UnixNano())
+	randIndex := rand.Intn(len(supportedRegions))
+	return supportedRegions[randIndex]
```
Comment (brikis98) [inline aws/eks_utils_for_test.go]: There's a terratest `GetRandomRegion` helper that takes in a whitelist and/or blacklist of regions you can use...

File: aws/eks.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+// getAllEksClusters returns a list of strings of EKS Cluster Names that uniquely identify each cluster.
+func getAllEksClusters(awsSession *session.Session, excludeAfter time.Time) ([]*string, error) {
```
Comment (brikis98) [inline aws/eks.go]: Don't you need to call this EKS code from somewhere? I think there's a struct of some sort you need to add and then some extra API calls in the main CLI app...

Comment (yorinasub17) [reply aws/eks.go]: Yup... I can't believe I missed that...

Comment (yorinasub17) [reply aws/eks.go]: I was so tied up in the tests and it was getting late that I got a bit too trigger happy I guess.

**Comment by yorinasub17** [comment]
One downside of this: the tests now have an extra 20 minutes because launching and tearing down EKS clusters take that long...

**Comment by brikis98** [comment]
> One downside of this: the tests now have an extra 20 minutes because launching and tearing down EKS clusters take that long...

Does it run in parallel with the other tests so the whole build takes 20 min? Or is it sequential?

**Comment by yorinasub17** [comment]
It runs parallel, so the whole build takes 20 minutes. For comparison, it was ~5 mins before.

**Comment by brikis98** [comment]
Understood. Not much we can do, TBH. Eventually, we'll want to nuke RDS, Elasticsearch, and other slow stuff too, so the build will only get slower. Such is DevOps.

**Comment by rfvermut** [comment]
Can anything be done to finish this?

**Comment by autero1** [comment]
We don't seem to have verbose output in the tests. Are there any particular tests that keep failing?

**Comment by tonerdo** [comment]
@autero1 see #50

**Comment by yorinasub17** [comment]
Thanks @tonerdo for the test fixes! I took a quick scan and I believe I addressed all the comments, so I am going to go ahead and merge this and release!

**Pull request merged**